test/common/api_utils.py

```diff
@@ -0,0 +1,170 @@
+from on_http_api2_0 import ApiApi as Api
+from config.api2_0_config import config
+from json import loads
+import sys
+import os
+
+
+def api_node_select_argsList(nodeType='compute', validate_obm=False):
+    """
+    This routine produces a list of node Id's that satisfy the requested parameters. 
+    Nodes may be filtered by Node Id,  SKU name, OBM MAC address which are defined
+    in the system environment.  The node type and OBM setting may be specified.
+    :param  nodeType: Node type (i.e. compute) (default= 'compute')
+    :param  validate_obm: should the node have OBM settings (default=False)
+    return: A list with node IDs that match node type, SKU, and possible valid OBM settings.
+    """
+    return api_node_select(config.api_client,
+                           nodeId=os.getenv("NODEID", "None"),
+                           skuName=os.getenv("SKU", "all"),
+                           obmmac=os.getenv("OBMMAC", "all"),
+                           nodeType=nodeType,
+                           validate_obm=validate_obm,
+                           verbosity=os.getenv("VERBOSITY", "0")
+                           )
+
+
+
+def api_node_select(client,
```

Comment:
I tried to read the whole implementation of this function and gives me a feeling of the logic is not so clear. Maybe I don't fully understand your design and maybe I don't understand all challenge you meet, if you have enough reason, I'm OK for your code.

I just thinking about if I want to implement this feature, I will take following steps (just some pseudocode):
```python
node_list = api.get_all_nodes();
filter_by_nodeId(node_list); // if any node doesn't meet the nodeId requirement, it will be removed from node_list
filter_by_skuName(node_list);  // if any node doesn't meet the skuName requirement, it will be removed from node_list
filter_by_nodeType(node_list);  // if any node doesn't meet the nodeType requirement, it will be removed from node_list
.... //filter by other criteria
//Until this step, the node_list stores all valid node
node_ids = extract_node_ids(node_list); //return only node id rather than the whole node document.
```